### ISSUE TITLE
Grails 3.0.8 hasMany/belongsTo validation and save issue

### ISSUE THREAD

**Issue by olliefreeman** [grails/grails-core#9308]
No idea why this is happening but when using `static hasMany = [belongs: Belong]` in one class and then `static belongsTo = [owner: Has]` there are 2 issues from the `grails.rest.RestfulController` and the resulting databind when posting what I believe to be an acceptable XML, as you can see no `id` as this won't be known, and no `owner` as this shouldn't be required.

```xml
<has>
    <belongs>
        <belong>
            <date>2015-10-01</date>
            <test>testing string</test>
        </belong>
    </belongs>
    <datetime>2015-09-30T10:23:53.539</datetime>
    <name>testing</name>
    <sample>a sample</sample>
</has>
```
The issues are:
1. Validation is not occurring on the belongs elements when calling validate in the `RestfulController.save()` method, which means we get a 500 internal server error when the instance.save() is called as the elements are not valid
2. The reason they are not valid is the owner element is null as it is not included in the XML. Surely during standard databinding it should handle the hasMany/belongsTo issue. The grails added `addToBelongs()` method does this automatically. Looking into the code it uses leftshift to add to collection, in this situation this probably isn't suitable. 

I have added an example to the following repo https://github.com/olliefreeman/grails-issues to reproduce the error. If this behaviour is as expected can you recommend a way round this, as it cascades down, so if i have hasMany/belongsTo inside a sub element I hit the same issue, and at this point I'm not sure what options I have wrt implementing binding classes etc to fix it.

**Comment by olliefreeman** [comment]
this also seems to be an issue with hasOne/belongsTo at least at the point when you construct an object using groovy's map argument constructor

```groovy
new TestObject(
                        metadata: new Metadata(
                                              date: DateTimeConverter.toDate(LocalDate.of(2015, 02, 02)),
                                               sourceOrganisation: 'dev'),
                        subject: new Subject(
                                details: new Details(
                                        sample: 'testing'
                                        eventDetails: new EventDetail(
                                                eventDate: LocalDate.of(2015, 02, 02),
                                                organisationId: 'dev_org',
                                                eventReference: 'dev_ref',
                                                primaryDiagnosis: 'dev_diag'
                                        )
                                )
                        
                         )
                )
```

where Subject has `static belongsTo = [owner: TestObject]` and TestObject has `static hasOne = [subject: Subject]`

I end up with Subject object with no owner set.

**Comment by olliefreeman** [comment]
and one-to-one relationships

**Comment by olliefreeman** [comment]
Fix #9312 should fix this issue
